src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/csra/CsraResourceIntTest.kt

```diff
@@ -0,0 +1,341 @@
+      fun `no assessment type`() {
+        webTestClient.post().uri("/csra/A1111AA")
+          .headers(setAuthorisation(roles = listOf("NOMIS_PRISONER_API__SYNCHRONISATION__RW")))
+          .contentType(MediaType.APPLICATION_JSON)
+          .body(
+            fromValue(
+              """{ 
+                "assessmentDate": "2025-12-14",
+                "score": "1200",
+                "status": "A",
+                "assessmentStaffId": ${staff.id},
+                "createdBy": "BILLSTAFF",
+                "type": "DUFF"
+              }""",
+            ),
+          )
+          .exchange()
+          .expectStatus().isBadRequest
+          .expectBody()
+          .jsonPath("userMessage").value<String> {
+            assertThat(it).contains("String \"DUFF\": not one of the values accepted for Enum class")
+          }
+      }
+
+      @Test
+      fun `no created by user`() {
+        webTestClient.post().uri("/csra/A1111AA")
+          .headers(setAuthorisation(roles = listOf("NOMIS_PRISONER_API__SYNCHRONISATION__RW")))
+          .contentType(MediaType.APPLICATION_JSON)
+          .body(
+            fromValue(
+              """{ 
+          "assessmentDate": "2025-12-14",
+          "type": "CSRF",
+          "score": "1200",
+          "status": "A",
+          "assessmentStaffId": ${staff.id}
+          }""",
+            ),
+          )
+          .exchange()
+          .expectStatus().isBadRequest
+          .expectBody()
+          .jsonPath("userMessage").isEqualTo("Bad request: createdBy field is required")
+      }
+
+      @Test
+      fun `invalid created by user`() {
+        webTestClient.post().uri("/csra/A1111AA")
+          .headers(setAuthorisation(roles = listOf("NOMIS_PRISONER_API__SYNCHRONISATION__RW")))
+          .contentType(MediaType.APPLICATION_JSON)
+          .body(
+            fromValue(
+              """{
+              "assessmentDate": "2025-12-14",
+              "type": "CSRF",
+              "score": "1200",
+              "status": "A",
+              "assessmentStaffId": ${staff.id},
+              "createdBy": "DUFF"
+               }""",
+            ),
+          )
+          .exchange()
+          .expectStatus().isBadRequest
+          .expectBody()
+          .jsonPath("userMessage").isEqualTo("Bad request: Cannot find user DUFF")
+      }
+    }
+
+    @Nested
+    inner class HappyPath {
+      @Test
+      fun `can create a CSRA with full data`() {
+        val created = webTestClient.post().uri("/csra/A1111AA")
+          .headers(setAuthorisation(roles = listOf("NOMIS_PRISONER_API__SYNCHRONISATION__RW")))
+          .contentType(MediaType.APPLICATION_JSON)
+          .body(fromValue(validFullCreateJsonRequest()))
+          .exchange()
+          .expectStatus().isOk
+          .expectBody<CsraCreateResponse>()
+          .returnResult()
+          .responseBody!!
+
+        val data = webTestClient.get().uri("/csra/booking/${created.bookingId}/sequence/${created.sequence}")
```

Comment:
I think in this case a repository lookup might be better than endpoint call since what we care about is the NOMIS data and how it is written